### DISCUSSION TITLE
which js and css to exclude from the delay

### DISCUSSION THREAD

**Discussion by Clearified** [wp-statistics/wp-statistics#417] [Q&A]
Hello,
1st I would like to thank the team for the hard work and the great plugin!
I would love to know which js/css to exclude from delay to keep the plugin functional as it must!
I have excluded chart.min.js & gulpfile.js to get the online users @ the Summery but I don't get search engines search..

Thank you in advance.

**Comment by mostafasoufi** [comment]
Hi,

Thank you for opening the issue, are you looking for to deregister some CSS/js from the plugin because of the page load? if you disable the chartsJs library, you won't have any charts in the overview.

Can I ask why you want to do that?

**Reply by Clearified** [reply]
Hello,
Thank you for the Reply!
As I mentioned that I am optimizing my WP website and the JS is delayed but an option to exclude from the delay is available and I want to exclude the JS/CSS that helps the plugin of showing the search engines results ( how many came from google, yandex...etc )

**Reply by mostafasoufi** [reply]
In the front, the plugin doesn't load any js or CSS. just there is an XHR request behind the page if you enabled the Cache Compatibility option.

https://github.com/wp-statistics/wp-statistics/blob/master/includes/class-wp-statistics-frontend.php#L69-L101
https://github.com/wp-statistics/wp-statistics/blob/master/includes/admin/templates/tracker-js.php

**Comment by Clearified** [comment]
Hello,
I have a js/css delay for optimizing my WP website, and it affected in the
performance of WP-Statistics so I have to add exclusion for the js/css
important ones that helps for showing the search details as for now it
works well but it doesn't show me who came from google and how many.

Regards,

**Comment by Clearified** [comment]
hello,
I have excluded the following from the js Delay function and everything is working BUT the Search Engine Referrals
analytics
chart.min.js
search.js
tinymce.min.js

May I know which js to exclude from the delay to have the plugin works well?

Regards,

**Reply by mostafasoufi** [reply]
The only js that the plugin needs is `chart.min.js`

**Comment by Clearified** [comment]
Thank you for the Reply!
Actually I don't know what makes it stop working suddenly..
BTW I use CloudFlare CDN. Is there any special settings for it?

**Reply by mostafasoufi** [reply]
Not really, there is no special settings for CloudFlare at the moment, but maybe later, we'll work on.

Best.
